instrumentation/spring/spring-webflux/spring-webflux-5.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/spring/webflux/v5_0/server/RouterFunctionInstrumentation.java

```diff
@@ -61,14 +62,17 @@ public void transform(TypeTransformer transformer) {
   @SuppressWarnings("unused")
   public static class RouteAdvice {
 
+    @AssignReturned.ToReturned
     @Advice.OnMethodExit(onThrowable = Throwable.class, suppress = Throwable.class)
-    public static void methodExit(
+    public static Mono<HandlerFunction<?>> methodExit(
         @Advice.This RouterFunction<?> thiz,
-        @Advice.Return(readOnly = false) Mono<HandlerFunction<?>> result,
+        @Advice.Return Mono<HandlerFunction<?>> originalResult,
         @Advice.Thrown Throwable throwable) {
+      Mono<HandlerFunction<?>> result = originalResult;
```

Comment:
Nit: no need for the intermediate variable if you directly return the new value here